src/cr/cube/matrix/cubemeasure.py

```diff
@@ -925,40 +1089,15 @@ def _array_type_std_res(self, counts, total, rowsum, colsum):
         variance = rowsum * colsum * (total - rowsum) * (total - colsum) / total ** 3
         return (counts - expected_counts) / np.sqrt(variance)
 
-    @classmethod
-    def _means_matrix_factory(cls, cube, dimensions, slice_idx):
-        """ -> matrix object appropriate to means `cube`."""
-        dimension_types = cube.dimension_types[-2:]
-
-        if dimension_types == (DT.MR, DT.MR):
-            # --- this MEANS_MR_X_MR case hasn't arisen yet ---
-            raise NotImplementedError(
-                "MR x MR with means is not implemented"
-            )  # pragma: no cover
-
-        MatrixCls = (
-            _NumArrayXMrMeansMatrix
-            if dimension_types == (DT.NUM_ARRAY, DT.MR)
-            else _NumArrayXCatMeansMatrix
-            if dimension_types[0] == DT.NUM_ARRAY
-            else _MrXCatMeansMatrix
-            if dimension_types[0] == DT.MR
-            else _CatXMrMeansMatrix
-            if dimension_types[1] == DT.MR
-            else _CatXCatMeansMatrix
-        )
-        counts, unweighted_counts = (
-            (cube.counts[slice_idx], cube.unweighted_counts[slice_idx])
-            if cube.ndim == 3
-            else (cube.counts, cube.unweighted_counts)
-        )
-        return MatrixCls(dimensions, counts, unweighted_counts)
-
     @staticmethod
     def _regular_matrix_class(dimension_types):
         """Return BaseCubeResultMatrix subclass appropriate to `dimension_types`."""
         return (
-            _MrXMrMatrix
+            _NumArrayXMrMatrix
+            if dimension_types == (DT.NUM_ARRAY, DT.MR)
+            else _NumArrayXCatMatrix
+            if dimension_types[0] == DT.NUM_ARRAY
+            else _MrXMrMatrix
             if dimension_types == (DT.MR, DT.MR)
             else _MrXCatMatrix
             if dimension_types[0] == DT.MR
```

Comment:
This is only called once and should be inlined. It made sense to be separate back when the factories for means and counts etc. were all together, but not anymore.